### PR TITLE
Hello world rest

### PR DESCRIPTION
**Pull request by miklosduma**
Doctests for Python REST example

### CODE REVIEW

File: docs/tutorial_rest_python.rst [hunk 1]
```diff
@@ -0,0 +1,212 @@
+#############
+Hello, Python
+#############
+
+In case you've already signed up for an Erlang course, by all means go for it.
+
+Just know that you have the option to write the operations' logic in Python or JavaScript.
+
+Let's see how we can use a Python module as the logic behind an SSE configuration.
```
Comment (lucafavatellawork) [inline docs/tutorial_rest_python.rst]: I had the impression that the Python source code was passed to the SSE node that was going to derive a service definition from that... so I propose to clarify upfront that the logic written in Python does not run on the SSE node but in the SSE CLI.
```suggestion
As you already know, the ``Expr`` service supports executing Erlang expressions on the SSE node.

Once connected to the SSE node, you can hook to it at runtime arbitrary operations' logic: such logic will not run on the SSE node but on your host (your host does not need to be publicly addressable).

The SSE CLI library makes it easy to write the operations' logic in Python.

Let's see how we can use a Python module as the logic behind an SSE configuration.
```

File: docs/tutorial_rest_python.rst [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+The configuration
+-----------------
+
+The markup of the `HelloWorldRest configuration <https://github.com/opensparkl/sse_cli/blob/master/docs/examples/hello-world-rest/hello-world-rest.xml>`_ is nearly identical to the one presented in the `HelloWorld3 <https://github.com/opensparkl/sse_cli/blob/master/docs/tutorial3.rst>`_ tutorial.
```
Comment (lucafavatellawork) [inline docs/tutorial_rest_python.rst]: I see. Analogously to https://github.com/opensparkl/sse_cli/pull/62/files#diff-e6d4018bc768a75662e2fe1ee9643987L34 : I see the `GesState` request has `name` field.

Comment (lucafavatellawork) [inline docs/tutorial_rest_python.rst]: I understand that each of these doctest tutorials puts exactly one configuration at the same configuration path, and the cleanup ensures deleting that. So what follows is question out of scope of this specific PR.

If configurations are placed on same node at different locations with same field names, what prevents a call to a solicit on a path to trigger request-reply etc. from the other configuration? (I expect there is some scoping with folders though may be not explained so far.)

File: docs/tutorial_rest_python.rst (hunk 1 above)
Comment (miklosduma) [reply docs/tutorial_rest_python.rst]: You're right the Python code runs on the "host" running the CLI command. It is indeed worth clarifying.

File: docs/tutorial_rest_python.rst (hunk 2 above)
Comment (miklosduma) [reply docs/tutorial_rest_python.rst]: Good point and good guess. The mix element (that is actually a special folder) acts as a boundary.

File: docs/tutorial_rest_python.rst [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+Have you heard about Chekhov's gun? Well, I have. I mentioned at the beginning that the ``Rest`` service is also specified as a client on the notify and solicit operations.
+
+Invoking client operations
+--------------------------
+
+When we executed the ``sparkl service`` command, it returned a ``Service`` object we bound to a variable called ``rest_service``.
+
+This ``Service`` object is associated with the ``Rest`` service in the SSE configuration. This service can trigger the notify or solicit operations it is a client on.
+
+When we use the ``sparkl call`` command to trigger a transaction via a solicit - we basically act as a client on that solicit.
```
Comment (lucafavatellawork) [inline docs/tutorial_rest_python.rst:111]: `sparkl call` does an HTTP POST request, right? I.e. no WebSocket.

File: docs/tutorial_rest_python.rst [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    >>> from queue import Queue
+    >>> msg_queue = Queue()
+    >>> events = sparkl('listen')
+    >>> def callback(response_struct):
+    ...     response = response_struct['response']
+    ...     greeting = response_struct['data']['greeting']
+    ...     msg_queue.put('Response: {}, greeting: {}'.format(response, greeting))
+    >>> rest_service.solicit(
+    ...     {
+    ...         'solicit': 'API/Say',
```
Comment (lucafavatellawork) [inline docs/tutorial_rest_python.rst:170]: In a real deployment (where the site does not run `sparkl('listen')` and does not wait for output), does the site have the guarantee that the SSE node processed the preceding `SetState`, triggered by the `SendState` notify, before the SSE node processes the `GetState`, triggered by this `Say` solicit?

If yes, whose responsibility is it to guarantee that?

File: docs/tutorial_rest_python.rst (hunk 3 above)
Comment (miklosduma) [reply docs/tutorial_rest_python.rst:111]: That's correct, yes

File: docs/tutorial_rest_python.rst (hunk 4 above)
Comment (miklosduma) [reply docs/tutorial_rest_python.rst:170]: It is the developer's responsibility. Notifies come back to you immediately so they don't wait for the outcome like a solicit does. This is meant to be an example, not a proof of concept :)

**Pull request merged**